src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -315,50 +316,464 @@ public static void Main()
                     // (10,36): error CS1003: Syntax error, ',' expected
                     //         _ = new C() is { Prop1[0]: {} };
                     Diagnostic(ErrorCode.ERR_SyntaxError, "{").WithArguments(",", "{").WithLocation(10, 36),
-                    // (11,26): error CS9000: Identifier or a simple member access expected.
+                    // (11,26): error CS8918: Identifier or a simple member access expected.
                     //         _ = new C() is { 1: {} };
                     Diagnostic(ErrorCode.ERR_InvalidNameInSubpattern, "1").WithLocation(11, 26));
         }
 
-        [Fact]
+        [Fact, WorkItem(52956, "https://github.com/dotnet/roslyn/issues/52956")]
         public void ExtendedPropertyPatterns_05()
         {
             var program = @"
 class C
 {
-    C Prop1, Prop2;
+    C Field1, Field2, Field3, Field4;
+    public void M()
+    {
+        _ = this is { Field1.Field2.Field3: {} };
+        _ = this is { Field4: {} };
```

Comment:
:memo: We previously didn't count this as a read on `Field4` (existing bug got fixed). Previously produced: `warning CS0169: The field 'C.Field4' is never used`